src/metabase/api/database.clj

```diff
@@ -881,7 +881,7 @@
 ;; Currently these match the titles of the admin UI buttons that call these endpoints
 
 ;; Should somehow trigger sync-database/sync-database!
-(api/defendpoint POST "/:id/sync_schema"
+(api/defendpoint ^{:content-types #{:content/*}} POST "/:id/sync_schema"
```

Comment:
For POST endpoints that don't expect a body does it make sense to validate the `Content-Type`? Maybe what we should actually be validating is that the body is empty or something

Reply:
Or maybe only validate Content-Type if body is present... it seems a little dangerous that we have to remember to put `^{:content-types #{:content/*}}` on every endpoint that doesn't expect a body now or risk breaking things for people

Reply:
Yeah i thought about that. Some issues: we've have to manually annotate which ones don't read the request which could get wild.
As for reading the request, I'm not sure that works. If it's json, we eagerly parse that in the middleware. For others, we kinda want the input stream to be read by the handler.

If this proves onerous, perhaps we could switch and annotate which ones we want to enforce have a content type rather than adding exceptions for ones we don't care about.

Reply:
This was an awesome thought actually. I have the args, and can get the args pulled from the route. If all args come from the route, then we can just ignore the content-type completely. Good thinking.